setup.go

```diff
@@ -362,52 +360,7 @@ func Setup(ctx context.Context, cfg Config, key crypto.PrivKey, dnsCache *cached
 }
 
 func setupDatastore(cfg Config) (datastore.Batching, error) {
-	badgerOpts := badger.DefaultOptions("")
-	badgerOpts.CompactL0OnClose = false
-	// ValueThreshold: defaults to 1MB! For us that means everything goes
-	// into the LSM tree and that means more stuff in memory.  We only
-	// put very small things on the LSM tree by default (i.e. a single
-	// CID).
-	badgerOpts.ValueThreshold = 256
-
-	// BlockCacheSize: instead of using blockstore, we cache things
-	// here. This only makes sense if using compression, according to
-	// docs.
-	badgerOpts.BlockCacheSize = cfg.InMemBlockCache // default 1 GiB.
-
-	// Compression: default. Trades reading less from disk for using more
-	// CPU. Given gateways are usually IO bound, I think we can make this
-	// trade.
-	if badgerOpts.BlockCacheSize == 0 {
-		badgerOpts.Compression = options.None
-	} else {
-		badgerOpts.Compression = options.Snappy
-	}
-
-	// If we write something twice, we do it with the same values so
-	// *shrugh*.
-	badgerOpts.DetectConflicts = false
-
-	// MemTableSize: Defaults to 64MiB which seems an ok amount to flush
-	// to disk from time to time.
-	badgerOpts.MemTableSize = 64 << 20
-	// NumMemtables: more means more memory, faster writes, but more to
-	// commit to disk if they get full. Default is 5.
-	badgerOpts.NumMemtables = 5
-
-	// IndexCacheSize: 0 means all in memory (default). All means indexes,
-	// bloom filters etc. Usually not huge amount of memory usage from
-	// this.
-	badgerOpts.IndexCacheSize = 0
-
-	opts := badger4.Options{
-		GcDiscardRatio: 0.3,
-		GcInterval:     20 * time.Minute,
-		GcSleep:        10 * time.Second,
-		Options:        badgerOpts,
-	}
-
-	return badger4.NewDatastore(filepath.Join(cfg.DataDir, "badger4"), &opts)
+	return flatfs.CreateOrOpen(filepath.Join(cfg.DataDir, "flatfs"), flatfs.NextToLast(2), true)
```

Comment:
```suggestion
	return flatfs.CreateOrOpen(filepath.Join(cfg.DataDir, "flatfs"), flatfs.NextToLast(3), false)
```

- Why would we benefit from `sync=true` when persistence doesn't really mater?
- @hsanjuan you've been running `NextToLast(3)` on most infra that uses FlatFS, right? My understanding is that at larger scale it's more effective.

Reply:
yes, helps keeping folder size smaller at larger scales, so safer in general. Sync destroys all performance.

Reply:
@aschmahmann I've actually been using both of your suggestions here in `kubo` for quite some time, so these changes work for me.